src/tempo.py

```diff
@@ -64,6 +60,21 @@ def _worker_config(self) -> Dict[str, Any]:
 
         return config
 
+    def restart(self):
+        """Override the worker's restart logic."""
+        # if we don't have the remote-write endpoints, don't try to restart the Tempo service
+        # as it will never start and we'll be stuck executing the hook.
+        # Exit and let collect-unit-status set the charm to blocked.
+        roles = self.roles
+        if len(roles) > 0 and "metrics-generator" in roles:
```

Comment:
```suggestion
        if roles and "metrics-generator" in roles:
```

or even (although perhaps a bit more cryptic):
> if "metrics-generator" in (roles or ()):